src/server/managers/AppListenerManager.ts

```diff
@@ -16,32 +17,46 @@ import {
 import { IUser } from '../../definition/users';
 import { MessageBuilder, MessageExtender, RoomBuilder, RoomExtender } from '../accessors';
 import { AppManager } from '../AppManager';
-import { AppInterface } from '../compiler';
 import { Message } from '../messages/Message';
 import { Utilities } from '../misc/Utilities';
 import { ProxiedApp } from '../ProxiedApp';
 import { Room } from '../rooms/Room';
 import { AppAccessorManager } from './AppAccessorManager';
 
+export class EssentialAppDisabledException extends AppsEngineException {}
+
 export class AppListenerManager {
     private am: AppAccessorManager;
     private listeners: Map<string, Array<string>>;
+    /**
+     * Blocked events are those who are listed in an app's
+     * "essentials" list but the app is disabled.
+     *
+     * They will throw a BlockedEventException upon call
+     */
+    private blockedEvents: Map<string, Array<string>>;
```

Comment:
```suggestion
    private blockedEvents: Map<string, Set<string>>;
```
Why not use set?

Reply:
It honestly did not cross my mind. Why would you consider using `Set` in this case?

Reply:
Given that the `Set` API provides a more convenient way to delete items, I think changing it makes sense